src/cocaine-app/planner/__init__.py

```diff
@@ -176,6 +189,93 @@ def _get_idle_groups(self, days_of_idle):
+        :return: dict ['couple_id'] = last_ttl_run_time
+        """
+        result = {}
+
+        couples_data = self.collection.find().sort('cleanup_ts', pymongo.ASCENDING)
+        if couples_data.count() < len(storage.replicas_groupsets):
+            logger.info('Sync cleanup data is required: {0} records/{1} couples'.format(
+                couples_data.count(), len(storage.replicas_groupsets)))
+            self.sync_historic_data(recover_ts=0, cleanup_ts=int(time.time()))
+            couples_data = self.collection.find().sort('cleanup_ts', pymongo.ASCENDING)
+
+        for couple_data in couples_data:
+            c = couple_data['couple']
+
+            if c == "None":
+                logger.error("Damaged couple? {}".format(couple_data))
+                continue
+
+            group_id = int(c.split(":")[0])
+
+            if group_id not in storage.groups:
+                logger.error("Not valid group is within collection {}".format(group_id))
+                continue
+            group = storage.groups[group_id]
+            if not group.couple:
+                logger.error("Group in collection is uncoupled {}".format(str(group)))
+                continue
+
+            ns_settings = self.namespaces_settings.get(group.couple.namespace.id)
+            if ns_settings and not ns_settings.attributes.ttl.enable:
+                logger.debug("Skipping group {} cause ns '{}' no ttl".format(group_id, group.couple.namespace.id))
+                continue
+
+            # if couple_data doesn't contain cleanup_ts field then cleanup_ts has never been run on this couple
+            # and None < idleness_threshold
+            result[group_id] = couple_data.get('cleanup_ts')
+
+        logger.info("History of last run is of {} len".format(len(result)))
+        return result
+
+    def _cleanup_aggregation_yt(self):
+        """
+        Cleanup YT aggregation table by removing outdated records (where outdated means their exp_date < ttl_last_run)
+        :return:
+        """
+
+        logger.info("Going to cleanup aggregation YT table")
+
+        # Configure parameters for work with YT
+        # XXX: when ttl_cleanup would be a separate class and a separate worker, remove re-reading params
+        # (the code is dup with get_yt_stat)
+        yt_cluster = self.params.get('ttl_cleanup', {}).get('yt_cluster', "")
+        yt_token = self.params.get('ttl_cleanup', {}).get('yt_token', "")
+        yt_attempts = self.params.get('ttl_cleanup', {}).get('yt_attempts', 3)
+        yt_delay = self.params.get('ttl_cleanup', {}).get('yt_delay', 10)
+
+        yt_wrapper = YqlWrapper(cluster=yt_cluster, token=yt_token, attempts=yt_attempts, delay=yt_delay)
+
+        aggregation_table = self.params.get('ttl_cleanup', {}).get('aggregation_table', "")
+
+        ttl_hist = self._get_ttl_cleanup_last_run()
+
+        yt_wrapper.cleanup_aggregaton_table(aggregate_table=aggregation_table, couples_hist=ttl_hist)
```

Comment:
Fixed in https://github.com/yandex/mastermind/pull/391/commits/85d74e4c3fd30346970906acc7028dc745e5831a